### PR TITLE
Fix Apple build errors

### PR DESCRIPTION
**Pull request by darrinsmart**
This fixes errors I encountered under OSX.

### CODE REVIEW

**Comment by legeana** [comment]
```
$ mesonwrap review libuv --pull-request=3 --approve --allow-no-license
Inspecting project libuv, branch 1.18.0.
Repo name valid: ☑
Branch name valid: ☑
Target branch is not master: ☑
Has readme.txt: ☑
Has LICENSE.build: ☒
Has upstream.wrap: ☑
Repo contains only buildsystem files: ☑
upstream.wrap has directory: ☑
upstream.wrap directory has no path separators: ☑
upstream.wrap has source_url: ☑
upstream.wrap has source_filename: ☑
upstream.wrap source_filename has no path separators: ☑
upstream.wrap has source_hash: ☑
upstream.wrap has source_url with version substring: ☑
Download URL works: ☑
Hash matches: ☑
upstream.wrap directory 'libuv-v1.18.0' exists: ☑
Patch merges with source: ☑
The Meson build system
Version: 0.54.2
Source dir: /tmp/tmpn78fzaov/src/libuv-v1.18.0
Build dir: /tmp/tmpn78fzaov/bin
Build type: native build
Project name: libuv
Project version: 1.18.0
C compiler for the host machine: cc (gcc 10.1.0 "cc (GCC) 10.1.0")
C linker for the host machine: cc ld.bfd 2.34.50.20200508
Host machine cpu family: x86_64
Host machine cpu: x86_64
Fetching value of define "__linux__" : 1 
Fetching value of define "__ANDROID__" :  
Fetching value of define "__APPLE__" :  
Run-time dependency threads found: YES
Library m found: YES
Library dl found: YES
Library util found: YES
Build targets in project: 2

Found ninja-1.10.0 at /usr/bin/ninja
meson setup: ☑
ninja: Entering directory `/tmp/tmpn78fzaov/bin'
[24/174] Compiling C object 'uv@sha/src_unix_linux-core.c.o'
../src/libuv-v1.18.0/src/unix/linux-core.c: In function ‘uv__io_poll’:
../src/libuv-v1.18.0/src/unix/linux-core.c:343:5: warning: converting a packed ‘struct uv__epoll_event’ pointer (alignment 1) to a ‘uv__io_t’ {aka ‘struct uv__io_s’} pointer (alignment 8) may result in an unaligned pointer value [-Waddress-of-packed-member]
  343 |     loop->watchers[loop->nwatchers] = (void*) events;
      |     ^~~~
In file included from ../src/libuv-v1.18.0/src/unix/internal.h:38,
                 from ../src/libuv-v1.18.0/src/unix/linux-core.c:27:
../src/libuv-v1.18.0/src/unix/linux-syscalls.h:90:8: note: defined here
   90 | struct uv__epoll_event {
      |        ^~~~~~~~~~~~~~~
In file included from ../src/libuv-v1.18.0/include/uv.h:62,
                 from ../src/libuv-v1.18.0/src/unix/linux-core.c:26:
../src/libuv-v1.18.0/include/uv-unix.h:93:8: note: defined here
   93 | struct uv__io_s {
      |        ^~~~~~~~
[145/174] Compiling C object 'uvtest@exe/test_test-thread.c.o'
In file included from ../src/libuv-v1.18.0/test/test-thread.c:23:
../src/libuv-v1.18.0/test/test-thread.c: In function ‘thread_check_stack’:
../src/libuv-v1.18.0/test/test-thread.c:220:15: warning: implicit declaration of function ‘pthread_getattr_np’; did you mean ‘pthread_attr_init’? [-Wimplicit-function-declaration]
  220 |   ASSERT(0 == pthread_getattr_np(pthread_self(), &attr));
      |               ^~~~~~~~~~~~~~~~~~
../src/libuv-v1.18.0/test/task.h:98:9: note: in definition of macro ‘ASSERT’
   98 |   if (!(expr)) {                                          \
      |         ^~~~
[173/174] Linking target uvtest
FAILED: uvtest 
cc  -o uvtest 'uvtest@exe/test_blackhole-server.c.o' 'uvtest@exe/test_dns-server.c.o' 'uvtest@exe/test_echo-server.c.o' 'uvtest@exe/test_run-tests.c.o' 'uvtest@exe/test_runner.c.o' 'uvtest@exe/test_test-active.c.o' 'uvtest@exe/test_test-async.c.o' 'uvtest@exe/test_test-async-null-cb.c.o' 'uvtest@exe/test_test-barrier.c.o' 'uvtest@exe/test_test-callback-order.c.o' 'uvtest@exe/test_test-callback-stack.c.o' 'uvtest@exe/test_test-close-fd.c.o' 'uvtest@exe/test_test-close-order.c.o' 'uvtest@exe/test_test-condvar.c.o' 'uvtest@exe/test_test-connection-fail.c.o' 'uvtest@exe/test_test-cwd-and-chdir.c.o' 'uvtest@exe/test_test-default-loop-close.c.o' 'uvtest@exe/test_test-delayed-accept.c.o' 'uvtest@exe/test_test-dlerror.c.o' 'uvtest@exe/test_test-eintr-handling.c.o' 'uvtest@exe/test_test-embed.c.o' 'uvtest@exe/test_test-emfile.c.o' 'uvtest@exe/test_test-env-vars.c.o' 'uvtest@exe/test_test-error.c.o' 'uvtest@exe/test_test-fail-always.c.o' 'uvtest@exe/test_test-fs-copyfile.c.o' 'uvtest@exe/test_test-fs-event.c.o' 'uvtest@exe/test_test-fs-poll.c.o' 'uvtest@exe/test_test-fs.c.o' 'uvtest@exe/test_test-fork.c.o' 'uvtest@exe/test_test-get-currentexe.c.o' 'uvtest@exe/test_test-get-loadavg.c.o' 'uvtest@exe/test_test-get-memory.c.o' 'uvtest@exe/test_test-get-passwd.c.o' 'uvtest@exe/test_test-getaddrinfo.c.o' 'uvtest@exe/test_test-gethostname.c.o' 'uvtest@exe/test_test-getnameinfo.c.o' 'uvtest@exe/test_test-getsockname.c.o' 'uvtest@exe/test_test-handle-fileno.c.o' 'uvtest@exe/test_test-homedir.c.o' 'uvtest@exe/test_test-hrtime.c.o' 'uvtest@exe/test_test-idle.c.o' 'uvtest@exe/test_test-ip4-addr.c.o' 'uvtest@exe/test_test-ip6-addr.c.o' 'uvtest@exe/test_test-ipc-send-recv.c.o' 'uvtest@exe/test_test-ipc.c.o' 'uvtest@exe/test_test-loop-handles.c.o' 'uvtest@exe/test_test-loop-alive.c.o' 'uvtest@exe/test_test-loop-close.c.o' 'uvtest@exe/test_test-loop-stop.c.o' 'uvtest@exe/test_test-loop-time.c.o' 'uvtest@exe/test_test-loop-configure.c.o' 'uvtest@exe/test_test-multiple-listen.c.o' 'uvtest@exe/test_test-mutexes.c.o' 'uvtest@exe/test_test-osx-select.c.o' 'uvtest@exe/test_test-pass-always.c.o' 'uvtest@exe/test_test-ping-pong.c.o' 'uvtest@exe/test_test-pipe-bind-error.c.o' 'uvtest@exe/test_test-pipe-connect-error.c.o' 'uvtest@exe/test_test-pipe-connect-multiple.c.o' 'uvtest@exe/test_test-pipe-connect-prepare.c.o' 'uvtest@exe/test_test-pipe-getsockname.c.o' 'uvtest@exe/test_test-pipe-pending-instances.c.o' 'uvtest@exe/test_test-pipe-sendmsg.c.o' 'uvtest@exe/test_test-pipe-server-close.c.o' 'uvtest@exe/test_test-pipe-close-stdout-read-stdin.c.o' 'uvtest@exe/test_test-pipe-set-non-blocking.c.o' 'uvtest@exe/test_test-pipe-set-fchmod.c.o' 'uvtest@exe/test_test-platform-output.c.o' 'uvtest@exe/test_test-poll.c.o' 'uvtest@exe/test_test-poll-close.c.o' 'uvtest@exe/test_test-poll-close-doesnt-corrupt-stack.c.o' 'uvtest@exe/test_test-poll-closesocket.c.o' 'uvtest@exe/test_test-poll-oob.c.o' 'uvtest@exe/test_test-process-title.c.o' 'uvtest@exe/test_test-queue-foreach-delete.c.o' 'uvtest@exe/test_test-ref.c.o' 'uvtest@exe/test_test-run-nowait.c.o' 'uvtest@exe/test_test-run-once.c.o' 'uvtest@exe/test_test-semaphore.c.o' 'uvtest@exe/test_test-shutdown-close.c.o' 'uvtest@exe/test_test-shutdown-eof.c.o' 'uvtest@exe/test_test-shutdown-twice.c.o' 'uvtest@exe/test_test-signal-multiple-loops.c.o' 'uvtest@exe/test_test-signal.c.o' 'uvtest@exe/test_test-socket-buffer-size.c.o' 'uvtest@exe/test_test-spawn.c.o' 'uvtest@exe/test_test-stdio-over-pipes.c.o' 'uvtest@exe/test_test-tcp-alloc-cb-fail.c.o' 'uvtest@exe/test_test-tcp-bind-error.c.o' 'uvtest@exe/test_test-tcp-bind6-error.c.o' 'uvtest@exe/test_test-tcp-close-accept.c.o' 'uvtest@exe/test_test-tcp-close-while-connecting.c.o' 'uvtest@exe/test_test-tcp-close.c.o' 'uvtest@exe/test_test-tcp-create-socket-early.c.o' 'uvtest@exe/test_test-tcp-connect-error-after-write.c.o' 'uvtest@exe/test_test-tcp-connect-error.c.o' 'uvtest@exe/test_test-tcp-connect-timeout.c.o' 'uvtest@exe/test_test-tcp-connect6-error.c.o' 'uvtest@exe/test_test-tcp-flags.c.o' 'uvtest@exe/test_test-tcp-open.c.o' 'uvtest@exe/test_test-tcp-read-stop.c.o' 'uvtest@exe/test_test-tcp-shutdown-after-write.c.o' 'uvtest@exe/test_test-tcp-unexpected-read.c.o' 'uvtest@exe/test_test-tcp-oob.c.o' 'uvtest@exe/test_test-tcp-write-to-half-open-connection.c.o' 'uvtest@exe/test_test-tcp-write-after-connect.c.o' 'uvtest@exe/test_test-tcp-writealot.c.o' 'uvtest@exe/test_test-tcp-write-fail.c.o' 'uvtest@exe/test_test-tcp-try-write.c.o' 'uvtest@exe/test_test-tcp-write-queue-order.c.o' 'uvtest@exe/test_test-thread-equal.c.o' 'uvtest@exe/test_test-thread.c.o' 'uvtest@exe/test_test-threadpool-cancel.c.o' 'uvtest@exe/test_test-threadpool.c.o' 'uvtest@exe/test_test-timer-again.c.o' 'uvtest@exe/test_test-timer-from-check.c.o' 'uvtest@exe/test_test-timer.c.o' 'uvtest@exe/test_test-tmpdir.c.o' 'uvtest@exe/test_test-tty.c.o' 'uvtest@exe/test_test-udp-alloc-cb-fail.c.o' 'uvtest@exe/test_test-udp-bind.c.o' 'uvtest@exe/test_test-udp-create-socket-early.c.o' 'uvtest@exe/test_test-udp-dgram-too-big.c.o' 'uvtest@exe/test_test-udp-ipv6.c.o' 'uvtest@exe/test_test-udp-multicast-interface.c.o' 'uvtest@exe/test_test-udp-multicast-interface6.c.o' 'uvtest@exe/test_test-udp-multicast-join.c.o' 'uvtest@exe/test_test-udp-multicast-join6.c.o' 'uvtest@exe/test_test-udp-multicast-ttl.c.o' 'uvtest@exe/test_test-udp-open.c.o' 'uvtest@exe/test_test-udp-options.c.o' 'uvtest@exe/test_test-udp-send-and-recv.c.o' 'uvtest@exe/test_test-udp-send-hang-loop.c.o' 'uvtest@exe/test_test-udp-send-immediate.c.o' 'uvtest@exe/test_test-udp-send-unreachable.c.o' 'uvtest@exe/test_test-udp-try-send.c.o' 'uvtest@exe/test_test-walk-handles.c.o' 'uvtest@exe/test_test-watcher-cross-stop.c.o' 'uvtest@exe/test_runner-unix.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libuv.so -pthread -lutil -Wl,--end-group '-Wl,-rpath,$ORIGIN/' -Wl,-rpath-link,/tmp/tmpn78fzaov/bin/
/usr/bin/ld: uvtest@exe/test_test-poll-closesocket.c.o:/tmp/tmpn78fzaov/bin/../src/libuv-v1.18.0/test/test-poll-closesocket.c:28: multiple definition of `sock'; uvtest@exe/test_test-poll-close-doesnt-corrupt-stack.c.o:/tmp/tmpn78fzaov/bin/../src/libuv-v1.18.0/test/test-poll-close-doesnt-corrupt-stack.c:35: first defined here
/usr/bin/ld: uvtest@exe/test_test-poll-closesocket.c.o:/tmp/tmpn78fzaov/bin/../src/libuv-v1.18.0/test/test-poll-closesocket.c:29: multiple definition of `handle'; uvtest@exe/test_test-poll-close-doesnt-corrupt-stack.c.o:/tmp/tmpn78fzaov/bin/../src/libuv-v1.18.0/test/test-poll-close-doesnt-corrupt-stack.c:36: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
ninja test: ☒
```

The original seems to be broken as well.
```
$ mesonwrap review libuv --branch=1.18.0 --allow-no-license
Inspecting project libuv, branch 1.18.0.
Repo name valid: ☑
Branch name valid: ☑
Target branch is not master: ☑
Has readme.txt: ☑
Has LICENSE.build: ☒
Has upstream.wrap: ☑
Repo contains only buildsystem files: ☑
upstream.wrap has directory: ☑
upstream.wrap directory has no path separators: ☑
upstream.wrap has source_url: ☑
upstream.wrap has source_filename: ☑
upstream.wrap source_filename has no path separators: ☑
upstream.wrap has source_hash: ☑
upstream.wrap has source_url with version substring: ☑
Download URL works: ☑
Hash matches: ☑
upstream.wrap directory 'libuv-v1.18.0' exists: ☑
Patch merges with source: ☑
The Meson build system
Version: 0.54.2
Source dir: /tmp/tmpshz9ats2/src/libuv-v1.18.0
Build dir: /tmp/tmpshz9ats2/bin
Build type: native build
Project name: libuv
Project version: 1.18.0
C compiler for the host machine: cc (gcc 10.1.0 "cc (GCC) 10.1.0")
C linker for the host machine: cc ld.bfd 2.34.50.20200508
Host machine cpu family: x86_64
Host machine cpu: x86_64
Fetching value of define "__ANDROID__" :  
Run-time dependency threads found: YES
Library m found: YES
Library dl found: YES
Library util found: YES
Build targets in project: 2

Found ninja-1.10.0 at /usr/bin/ninja
meson setup: ☑
ninja: Entering directory `/tmp/tmpshz9ats2/bin'
[24/174] Compiling C object 'uv@sha/src_unix_linux-core.c.o'
../src/libuv-v1.18.0/src/unix/linux-core.c: In function ‘uv__io_poll’:
../src/libuv-v1.18.0/src/unix/linux-core.c:343:5: warning: converting a packed ‘struct uv__epoll_event’ pointer (alignment 1) to a ‘uv__io_t’ {aka ‘struct uv__io_s’} pointer (alignment 8) may result in an unaligned pointer value [-Waddress-of-packed-member]
  343 |     loop->watchers[loop->nwatchers] = (void*) events;
      |     ^~~~
In file included from ../src/libuv-v1.18.0/src/unix/internal.h:38,
                 from ../src/libuv-v1.18.0/src/unix/linux-core.c:27:
../src/libuv-v1.18.0/src/unix/linux-syscalls.h:90:8: note: defined here
   90 | struct uv__epoll_event {
      |        ^~~~~~~~~~~~~~~
In file included from ../src/libuv-v1.18.0/include/uv.h:62,
                 from ../src/libuv-v1.18.0/src/unix/linux-core.c:26:
../src/libuv-v1.18.0/include/uv-unix.h:93:8: note: defined here
   93 | struct uv__io_s {
      |        ^~~~~~~~
[145/174] Compiling C object 'uvtest@exe/test_test-thread.c.o'
In file included from ../src/libuv-v1.18.0/test/test-thread.c:23:
../src/libuv-v1.18.0/test/test-thread.c: In function ‘thread_check_stack’:
../src/libuv-v1.18.0/test/test-thread.c:220:15: warning: implicit declaration of function ‘pthread_getattr_np’; did you mean ‘pthread_attr_init’? [-Wimplicit-function-declaration]
  220 |   ASSERT(0 == pthread_getattr_np(pthread_self(), &attr));
      |               ^~~~~~~~~~~~~~~~~~
../src/libuv-v1.18.0/test/task.h:98:9: note: in definition of macro ‘ASSERT’
   98 |   if (!(expr)) {                                          \
      |         ^~~~
[173/174] Linking target uvtest
FAILED: uvtest 
cc  -o uvtest 'uvtest@exe/test_blackhole-server.c.o' 'uvtest@exe/test_dns-server.c.o' 'uvtest@exe/test_echo-server.c.o' 'uvtest@exe/test_run-tests.c.o' 'uvtest@exe/test_runner.c.o' 'uvtest@exe/test_test-active.c.o' 'uvtest@exe/test_test-async.c.o' 'uvtest@exe/test_test-async-null-cb.c.o' 'uvtest@exe/test_test-barrier.c.o' 'uvtest@exe/test_test-callback-order.c.o' 'uvtest@exe/test_test-callback-stack.c.o' 'uvtest@exe/test_test-close-fd.c.o' 'uvtest@exe/test_test-close-order.c.o' 'uvtest@exe/test_test-condvar.c.o' 'uvtest@exe/test_test-connection-fail.c.o' 'uvtest@exe/test_test-cwd-and-chdir.c.o' 'uvtest@exe/test_test-default-loop-close.c.o' 'uvtest@exe/test_test-delayed-accept.c.o' 'uvtest@exe/test_test-dlerror.c.o' 'uvtest@exe/test_test-eintr-handling.c.o' 'uvtest@exe/test_test-embed.c.o' 'uvtest@exe/test_test-emfile.c.o' 'uvtest@exe/test_test-env-vars.c.o' 'uvtest@exe/test_test-error.c.o' 'uvtest@exe/test_test-fail-always.c.o' 'uvtest@exe/test_test-fs-copyfile.c.o' 'uvtest@exe/test_test-fs-event.c.o' 'uvtest@exe/test_test-fs-poll.c.o' 'uvtest@exe/test_test-fs.c.o' 'uvtest@exe/test_test-fork.c.o' 'uvtest@exe/test_test-get-currentexe.c.o' 'uvtest@exe/test_test-get-loadavg.c.o' 'uvtest@exe/test_test-get-memory.c.o' 'uvtest@exe/test_test-get-passwd.c.o' 'uvtest@exe/test_test-getaddrinfo.c.o' 'uvtest@exe/test_test-gethostname.c.o' 'uvtest@exe/test_test-getnameinfo.c.o' 'uvtest@exe/test_test-getsockname.c.o' 'uvtest@exe/test_test-handle-fileno.c.o' 'uvtest@exe/test_test-homedir.c.o' 'uvtest@exe/test_test-hrtime.c.o' 'uvtest@exe/test_test-idle.c.o' 'uvtest@exe/test_test-ip4-addr.c.o' 'uvtest@exe/test_test-ip6-addr.c.o' 'uvtest@exe/test_test-ipc-send-recv.c.o' 'uvtest@exe/test_test-ipc.c.o' 'uvtest@exe/test_test-loop-handles.c.o' 'uvtest@exe/test_test-loop-alive.c.o' 'uvtest@exe/test_test-loop-close.c.o' 'uvtest@exe/test_test-loop-stop.c.o' 'uvtest@exe/test_test-loop-time.c.o' 'uvtest@exe/test_test-loop-configure.c.o' 'uvtest@exe/test_test-multiple-listen.c.o' 'uvtest@exe/test_test-mutexes.c.o' 'uvtest@exe/test_test-osx-select.c.o' 'uvtest@exe/test_test-pass-always.c.o' 'uvtest@exe/test_test-ping-pong.c.o' 'uvtest@exe/test_test-pipe-bind-error.c.o' 'uvtest@exe/test_test-pipe-connect-error.c.o' 'uvtest@exe/test_test-pipe-connect-multiple.c.o' 'uvtest@exe/test_test-pipe-connect-prepare.c.o' 'uvtest@exe/test_test-pipe-getsockname.c.o' 'uvtest@exe/test_test-pipe-pending-instances.c.o' 'uvtest@exe/test_test-pipe-sendmsg.c.o' 'uvtest@exe/test_test-pipe-server-close.c.o' 'uvtest@exe/test_test-pipe-close-stdout-read-stdin.c.o' 'uvtest@exe/test_test-pipe-set-non-blocking.c.o' 'uvtest@exe/test_test-pipe-set-fchmod.c.o' 'uvtest@exe/test_test-platform-output.c.o' 'uvtest@exe/test_test-poll.c.o' 'uvtest@exe/test_test-poll-close.c.o' 'uvtest@exe/test_test-poll-close-doesnt-corrupt-stack.c.o' 'uvtest@exe/test_test-poll-closesocket.c.o' 'uvtest@exe/test_test-poll-oob.c.o' 'uvtest@exe/test_test-process-title.c.o' 'uvtest@exe/test_test-queue-foreach-delete.c.o' 'uvtest@exe/test_test-ref.c.o' 'uvtest@exe/test_test-run-nowait.c.o' 'uvtest@exe/test_test-run-once.c.o' 'uvtest@exe/test_test-semaphore.c.o' 'uvtest@exe/test_test-shutdown-close.c.o' 'uvtest@exe/test_test-shutdown-eof.c.o' 'uvtest@exe/test_test-shutdown-twice.c.o' 'uvtest@exe/test_test-signal-multiple-loops.c.o' 'uvtest@exe/test_test-signal.c.o' 'uvtest@exe/test_test-socket-buffer-size.c.o' 'uvtest@exe/test_test-spawn.c.o' 'uvtest@exe/test_test-stdio-over-pipes.c.o' 'uvtest@exe/test_test-tcp-alloc-cb-fail.c.o' 'uvtest@exe/test_test-tcp-bind-error.c.o' 'uvtest@exe/test_test-tcp-bind6-error.c.o' 'uvtest@exe/test_test-tcp-close-accept.c.o' 'uvtest@exe/test_test-tcp-close-while-connecting.c.o' 'uvtest@exe/test_test-tcp-close.c.o' 'uvtest@exe/test_test-tcp-create-socket-early.c.o' 'uvtest@exe/test_test-tcp-connect-error-after-write.c.o' 'uvtest@exe/test_test-tcp-connect-error.c.o' 'uvtest@exe/test_test-tcp-connect-timeout.c.o' 'uvtest@exe/test_test-tcp-connect6-error.c.o' 'uvtest@exe/test_test-tcp-flags.c.o' 'uvtest@exe/test_test-tcp-open.c.o' 'uvtest@exe/test_test-tcp-read-stop.c.o' 'uvtest@exe/test_test-tcp-shutdown-after-write.c.o' 'uvtest@exe/test_test-tcp-unexpected-read.c.o' 'uvtest@exe/test_test-tcp-oob.c.o' 'uvtest@exe/test_test-tcp-write-to-half-open-connection.c.o' 'uvtest@exe/test_test-tcp-write-after-connect.c.o' 'uvtest@exe/test_test-tcp-writealot.c.o' 'uvtest@exe/test_test-tcp-write-fail.c.o' 'uvtest@exe/test_test-tcp-try-write.c.o' 'uvtest@exe/test_test-tcp-write-queue-order.c.o' 'uvtest@exe/test_test-thread-equal.c.o' 'uvtest@exe/test_test-thread.c.o' 'uvtest@exe/test_test-threadpool-cancel.c.o' 'uvtest@exe/test_test-threadpool.c.o' 'uvtest@exe/test_test-timer-again.c.o' 'uvtest@exe/test_test-timer-from-check.c.o' 'uvtest@exe/test_test-timer.c.o' 'uvtest@exe/test_test-tmpdir.c.o' 'uvtest@exe/test_test-tty.c.o' 'uvtest@exe/test_test-udp-alloc-cb-fail.c.o' 'uvtest@exe/test_test-udp-bind.c.o' 'uvtest@exe/test_test-udp-create-socket-early.c.o' 'uvtest@exe/test_test-udp-dgram-too-big.c.o' 'uvtest@exe/test_test-udp-ipv6.c.o' 'uvtest@exe/test_test-udp-multicast-interface.c.o' 'uvtest@exe/test_test-udp-multicast-interface6.c.o' 'uvtest@exe/test_test-udp-multicast-join.c.o' 'uvtest@exe/test_test-udp-multicast-join6.c.o' 'uvtest@exe/test_test-udp-multicast-ttl.c.o' 'uvtest@exe/test_test-udp-open.c.o' 'uvtest@exe/test_test-udp-options.c.o' 'uvtest@exe/test_test-udp-send-and-recv.c.o' 'uvtest@exe/test_test-udp-send-hang-loop.c.o' 'uvtest@exe/test_test-udp-send-immediate.c.o' 'uvtest@exe/test_test-udp-send-unreachable.c.o' 'uvtest@exe/test_test-udp-try-send.c.o' 'uvtest@exe/test_test-walk-handles.c.o' 'uvtest@exe/test_test-watcher-cross-stop.c.o' 'uvtest@exe/test_runner-unix.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group libuv.so -pthread -lutil -Wl,--end-group '-Wl,-rpath,$ORIGIN/' -Wl,-rpath-link,/tmp/tmpshz9ats2/bin/
/usr/bin/ld: uvtest@exe/test_test-poll-closesocket.c.o:/tmp/tmpshz9ats2/bin/../src/libuv-v1.18.0/test/test-poll-closesocket.c:28: multiple definition of `sock'; uvtest@exe/test_test-poll-close-doesnt-corrupt-stack.c.o:/tmp/tmpshz9ats2/bin/../src/libuv-v1.18.0/test/test-poll-close-doesnt-corrupt-stack.c:35: first defined here
/usr/bin/ld: uvtest@exe/test_test-poll-closesocket.c.o:/tmp/tmpshz9ats2/bin/../src/libuv-v1.18.0/test/test-poll-closesocket.c:29: multiple definition of `handle'; uvtest@exe/test_test-poll-close-doesnt-corrupt-stack.c.o:/tmp/tmpshz9ats2/bin/../src/libuv-v1.18.0/test/test-poll-close-doesnt-corrupt-stack.c:36: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
ninja test: ☒
```

How did you test it?

**Comment by darrinsmart** [comment]
Thanks for the reply. I'm new to meson so I apologize if I've made a procedural error.

I tested the change by taking the libuv subproject directory from my project and building from there:

```
[darrin@n0 libuv-v1.18.0]$ meson /tmp/testbuild
The Meson build system
Version: 0.49.2
Source dir: /tmp/libuv-v1.18.0
Build dir: /tmp/testbuild
Build type: native build
Project name: libuv
Project version: 1.18.0
Native C compiler: cc (gcc 8.3.1 "cc (GCC) 8.3.1 20190507 (Red Hat 8.3.1-4)")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Fetching value of define "__linux__" : 1
Fetching value of define "__ANDROID__" : 
Fetching value of define "__APPLE__" : 
Dependency threads found: YES 
Library m found: YES
Library dl found: YES
Library util found: YES
Build targets in project: 2
Found ninja-1.8.2 at /usr/bin/ninja
[darrin@n0 libuv-v1.18.0]$ ninja -C /tmp/testbuild
ninja: Entering directory `/tmp/testbuild'
[151/173] Compiling C object 'uvtest@exe/test_test-thread.c.o'.
In file included from ../libuv-v1.18.0/test/test-thread.c:23:
../libuv-v1.18.0/test/test-thread.c: In function ‘thread_check_stack’:
../libuv-v1.18.0/test/test-thread.c:220:15: warning: implicit declaration of function ‘pthread_getattr_np’; did you mean ‘pthread_attr_init’? [-Wimplicit-function-declaration]
   ASSERT(0 == pthread_getattr_np(pthread_self(), &attr));
               ^~~~~~~~~~~~~~~~~~
../libuv-v1.18.0/test/task.h:98:9: note: in definition of macro ‘ASSERT’
   if (!(expr)) {                                          \
         ^~~~
[173/173] Linking target uvtest.
```
I did this on OSX (Catalina) and CentOS 8.

**Comment by darrinsmart** [comment]
I think this problem is due to a change in gcc 10.

It's already fixed in recent libuv releases: https://github.com/libuv/libuv/issues/2603

Adding the `-fcommon` flag should do as a workaround.

It's an independent issue really, not part of this PR, but I'm happy to roll in the change if you'd like.

**Comment by legeana** [comment]
No, you did everything correctly. However I am a maintainer of wrapdb itself, not each individual package so I have to ask for help from the wrap contributor who actually knows how the package works if something is broken.

I tried `-fcommon` but it didn't work. I would like to fix this issue first in order to be able to verify that your change doesn't actually break the Linux branch. Currently I can't do that unfortunately.

Could you please take a look at https://github.com/mesonbuild/libuv/pull/4? I tried applying the fix you proposed but it didn't seem to help.

File: meson.build [hunk 1]
```diff
@@ -43,6 +50,19 @@ if cc.get_define('__ANDROID__') != ''
              'src/unix/pthread-fixes.c' ]
 endif
 
+# OSX systems
+if cc.get_define('__APPLE__') != ''
+    uvsrc += [
+    	'src/unix/bsd-ifaddrs.c',
+	'src/unix/darwin.c',
+	'src/unix/darwin-proctitle.c',
```
Comment (legeana) [inline meson.build]: This looks like a mix of tabs and spaces, please `:retab`.

**Review by legeana — CHANGES_REQUESTED** [review]
Please rebase, I merged #4.

File: meson.build (hunk 1 above)
Comment (darrinsmart) [reply meson.build]: Should be tab-free now.

**Pull request merged**